### PR TITLE
Run box integration tests on Jenkins

### PR DESCRIPTION
**Pull request by jgogstad**
This PR adds a service account that Jenkins can use to run the Box integration tests. Please give me your Box usernames, and I'll give you access to it. Might be handy to see what files are in the account's Box if anything breaks.

~~I couldn't get this `new BoxFolder(api, "some-folder")` to work. The Box API consistently gives me 404. If I traverse by starting at `BoxFolder.getRootFolder()` it works fine (like [here](https://github.com/fs2-blobstore/fs2-blobstore/pull/50/files#diff-7e0ac39c06ae5eddf1f34595e9ae3793R55)).~~

~~I figured that not having this "absRoot" concept might ultimately be a good thing. Just have one way of specifying paths, and it will be consistent across all stores, see brief discussion on the Sftp `absRoot` in https://github.com/fs2-blobstore/fs2-blobstore/issues/15~~

~~WDYT?~~

### CODE REVIEW

**Comment by jgogstad** [comment]
One concern I have though is that the integration tests takes close to 7 minutes to run on my computer. I hope this is due to natural slowness on the Box side, but we should try to get some benchmarks here.

**Comment by gafiatulin** [comment]
> I couldn't get this new BoxFolder(api, "some-folder") to work. The Box API consistently gives me 404. 

It requires `folderId`, not folder name.

**Comment by jgogstad** [comment]
Added back `rootFolderId` parameter

File: box/src/main/scala/blobstore/box/BoxStore.scala [hunk 1]
```diff
@@ -18,17 +18,17 @@ package box
...
 import com.box.sdk.{BoxAPIConnection, BoxFile, BoxFolder, BoxItem, BoxResource}
 import fs2.{Pipe, Stream}
 
 import scala.jdk.CollectionConverters._
 
 final class BoxStore[F[_]](
   api: BoxAPIConnection,
-  rootFolderId: String,
   blocker: Blocker,
+  rootFolderId: String,
```
Comment (gafiatulin) [inline box/src/main/scala/blobstore/box/BoxStore.scala]: Revert to old order?

File: box/src/main/scala/blobstore/box/BoxStore.scala [hunk 2]
```diff
@@ -222,9 +222,11 @@ final class BoxStore[F[_]](
...
   def apply[F[_]](
     api: BoxAPIConnection,
-    rootFolderId: String,
-    blocker: Blocker
-  )(implicit F: Concurrent[F], CS: ContextShift[F]): BoxStore[F] = new BoxStore(api, rootFolderId, blocker)
+    blocker: Blocker,
+    rootFolderId: String = RootFolderId
+  )(implicit F: Concurrent[F], CS: ContextShift[F]): BoxStore[F] = new BoxStore(api, blocker, rootFolderId)
+
+  val RootFolderId = "0"
```
Comment (gafiatulin) [inline box/src/main/scala/blobstore/box/BoxStore.scala]: Is this needed?

Comment (gafiatulin) [reply box/src/main/scala/blobstore/box/BoxStore.scala]: Or is it being resolved to root folder?

Comment (jgogstad) [reply box/src/main/scala/blobstore/box/BoxStore.scala]: Yes, defaulting it to root if it's not set by the user. Changed the order of arguments to be consistent with the apply method where this is last due to a default parameter.

**Pull request merged**